### PR TITLE
#452: Review and use TTL memcache instead of clean manually

### PR DESCRIPTION
**Pull request by hieudinhngoc-ncc**
# Ticket
https://github.com/mezonai/mmn/issues/452

# General
- [x] No unnecessary files, debug logs, or commented code included.
- [ ] PR has at least 1 peer approval before senior review.
- [x] I fully understand the code I submitted.
- [x] No hardcoded configuration or secrets.
- [ ] **Environment Variables** changes? (Leave unchecked if N/A)

# Design & Solution Clarity
- This is refactor task

Design link (diagram / doc):
- This is refactor task

# Testing
- [x] Unit tests cover main logic and important edge cases.

> Note: Complex logic must not rely on manual testing only.

# UI evidence (For frontend only)
`<paste screenshots here>`

# Peer Review Checklist
- [ ] I understand the overall solution and flow.
- [ ] The code follows the provided design diagram.
- [ ] No obvious logic or security issues found.
- [ ] Tests are sufficient for the scope of this change.

### CODE REVIEW

File: mempool/dedup_service.go [hunk 1]
```diff
@@ -81,64 +77,25 @@ func (ds *DedupService) LoadTxHashes(latestSlot uint64) {
...
-	se.mu.Unlock()
-
+func (ds *DedupService) Add(txDedupHashes []string) {
 	for _, txDedupHash := range txDedupHashes {
-		shardIndex := getShardIndex(txDedupHash)
-		shard := ds.txShards[shardIndex]
-		shard.mu.Lock()
-		delete(shard.txDedupHashSet, txDedupHash)
-		shard.mu.Unlock()
+		ds.txDedupHashesCache.Set(txDedupHash, []byte{1})
```
Comment (TrongNgoVan) [inline mempool/dedup_service.go]: có nên log error cho mấy đoạn set&get này k o nhỉ?

File: transaction/tracker.go [hunk 2]
```diff
@@ -66,7 +76,7 @@ func (t *TransactionTracker) TrackProcessingTransaction(tx *Transaction) {
 func (t *TransactionTracker) RemoveTransaction(txHash string) {
 	txInterface, exists := t.processingTxs.LoadAndDelete(txHash)
 	if !exists {
-		t.markRemoved(txHash)
+		t.removedTxCache.Set(txHash, []byte{1})
```
Comment (TrongNgoVan) [inline transaction/tracker.go]: có nên log error cho mấy đoạn set&get này k o nhỉ?

File: zkverify/verify.go [hunk 3]
```diff
@@ -79,17 +68,18 @@ func makeCacheKey(sender, pubKey, proofB64, pubB64 string) string {
...
 	}
 
 	result := v.verifyInternal(sender, pubKey, proofB64, pubB64)
 
-	v.cache.Store(cacheKey, CacheEntry{
-		value:    result,
-		expireAt: time.Now().Add(CacheExpiryDuration),
-	})
+	resultBytes, _ := json.Marshal(result)
+	v.zkCache.Set(cacheKey, resultBytes)
```
Comment (TrongNgoVan) [inline zkverify/verify.go]: có nên log error cho mấy đoạn set&get này k o nhỉ?

Comment (hieudinhngoc-ncc) [reply zkverify/verify.go]: Đã add log error cho set, còn lúc get thì k cần thiết lắm, vì nếu get lỗi thì sẽ chạy verify bình thường và set lại cache => không ảnh hưởng gì

**Review by TrongNgoVan — APPROVED** [review]

File: transaction/tracker.go [hunk 4]
```diff
@@ -22,19 +20,30 @@ type TransactionTracker struct {
 	// processingTxs maps transaction hash to transaction
 	processingTxs sync.Map
 
-	historyList sync.Map
-
-	stopCh chan struct{}
+	removedTxCache *bigcache.BigCache
```
Comment (hoangdohuy) [inline transaction/tracker.go:23]: hmm consider correctness of logic when we use cache for blacklist

Comment (hoangdohuy) [reply transaction/tracker.go:23]: let me check more detail hmm

**Review by hoangdohuy — APPROVED** [review]
LGTM
